src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -27,22 +29,30 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/login";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    boolean loginStatus = userService.isUserLoggedIn();
+    String urlToRedirect = "/index.html";
 
-      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    ArrayList<String> userDetails = new ArrayList();
+    if (loginStatus) {
+      userDetails.add("true");
+      String userEmail = userService.getCurrentUser().getEmail();
+      String logoutURL = userService.createLogoutURL(urlToRedirect);
+      userDetails.add(
+          "<p>Hello, "
```

Comment:
Also, generally the server doesn't specify strings like this. After reading your JS below it seems like it would be much more appropriate to just return the userEmail and any other data inside a json object and have the JS construct the string to show to the user.